### PR TITLE
Add redux middlewate

### PR DESCRIPTION
**Pull request by ZorianaPro**

### CODE REVIEW

**Comment by ZorianaPro** [comment]
Hey HP,
so my PR is a big bigger than just a question that I have.
I am struggling with error handling after I call my API.
For example, lets look at this router `src/server/routes.js:28`. 
If there was an error creating something in DB I catch that error. Then in my component (I will make it a service, that will not stay in a component) when I fetch it, I want to catch the error that I return and display it for the user. But the problem is:
on the server log it throws one error message `InternalServerError: There was a duplicate key error` and in component it shows `Failed to fetch`.
Maybe I should do it completely different on the server as well?
HEEEEEELP!

File: src/server/routes.js [hunk 1]
```diff
@@ -6,40 +6,54 @@ const Member = require('./controllers/member');
...
+		const content = ctx.request.body;
+		const idea = await Member.addNewMember(content.id, content.member);
+		ctx.response.status = HttpStatus.OK;
+		ctx.response.body = idea.toJSON();
+	} catch (err) {
+		ctx.throw(HttpStatus.INTERNAL_SERVER_ERROR, JSON.stringify({message: err.message}));
+	}
 });
 
 router.get('/ideas/delete', async (ctx, next) => {
```
Comment (haensl) [inline src/server/routes.js:50]: The routes again ;)

This one should be `router.delete('/ideas')`. This means: The HTTP verb is [`DELETE`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE), the _resource_ you want to delete is `/ideas`.

The corresponding request from the client looks like this:

`DELETE /ideas`

So, in the end the schema boils down to:

`HTTP verb /resourceClass/individualID`

Examples:

* Retrieve all ideas: `GET /ideas`
* Retrieve a specific idea with id `1`: `GET /ideas/1`
* Delete idea with id `1`: `DELETE /ideas/1`
* Create an idea: `POST /ideas` _(with idea content in `body` as `JSON` as you already do)_
* Update idea with id `1`: `PUT /ideas/1` with updated idea (the whole (!) new object) in `body` as `JSON`
* Delete all ideas: `DELETE /ideas`

File: src/components/AddNewIdea/index.js [hunk 2]
```diff
@@ -73,20 +73,18 @@ const AddNewIdea = ({
...
+					setShowThankYou(true);
+					setTimeout(() => {
+						dispatch(close());
+						setShowThankYou(false);
+					}, 2000)
+				}
 			})
-			.then(() => {
+			.catch((err) => {
+				dispatch(error(err.message));
```
[the remark this answers is not in the document]
Comment (ZorianaPro) [reply src/components/AddNewIdea/index.js:86]: okay. so I fixed everything that you pointed to, and I have notoced than the problem is: `Access to fetch at 'http://localhost:8082/members' from origin 'http://localhost:3001' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.` in console. That's why I have `Failed to fetch` error caught here. But idea is still created and request was successful. Do you know how to fix that? Or I will fix that on production when I set everything up correct? @haensl

[the remark this answers is not in the document]
Comment (haensl) [reply src/components/AddNewIdea/index.js:86]: Like the error says: there are no CORS headers set by your Koa server. You can add those: [@koa/cors](https://www.npmjs.com/package/@koa/cors)

[the remark this answers is not in the document]
Comment (ZorianaPro) [reply src/components/AddNewIdea/index.js:86]: Yes, and I have it)))) 
I dont uderstand

[the remark this answers is not in the document]
Comment (haensl) [reply src/components/AddNewIdea/index.js:86]: You need to tell `fetch` to use `cors` mode, too ;)

**Pull request merged**